subprojects/core-api/src/main/java/org/gradle/api/artifacts/repositories/HttpHeaderCredentials.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.artifacts.repositories;
+
+import org.gradle.api.Incubating;
+
+/**
+ * Credentials that can be used to login to a protected server, e.g. a remote repository by using HTTP header.
+ *
+ * @since 4.10
+ */
+@Incubating
+public interface HttpHeaderCredentials extends org.gradle.api.credentials.HttpHeaderCredentials {
```

Comment:
@oehme Do you know why there are two of these interfaces for `PasswordCredentials` but not for `AwsCredentials`? I'm guessing the former was done for backwards compatibility so we don't need two here, do we?

Reply:
I strongly assume this was just for backwards compat.

Reply:
ok, so `org.gradle.api.artifacts.repositories.HttpHeaderCredentials` can be removed completely?

Should `DefaultPasswordCredentials` and `DefaultHttpHeaderCredentials` be moved from `org.gradle.api.internal.artifacts.repositories` to `org.gradle.internal.credentials`, where `DefaultAwsCredentials` is?

Reply:
> ok, so `org.gradle.api.artifacts.repositories.HttpHeaderCredentials` can be removed completely?

Yes, I'd say so until proven otherwise. 🙂 

> Should `DefaultPasswordCredentials` and `DefaultHttpHeaderCredentials` be moved from `org.gradle.api.internal.artifacts.repositories` to `org.gradle.internal.credentials`, where `DefaultAwsCredentials` is?

Yes, please! 👍

Reply:
Ok, I'll do this.
I updated the user guide already. Is there any other place/reference which has to be updated?

Reply:
Please create XML files for `HttpHeaderCredentials` and `HttpHeaderAuthentication` in `subprojects/docs/src/docs/dsl/` (see the one for `AwsCredentials` for an example) and add them to the appropriate places in this block:
https://github.com/gradle/gradle/blob/595dc5b6cee637ccae78e7a2364e79366ac073fd/subprojects/docs/src/docs/dsl/dsl.xml#L310-L340

Moreover, I think we should move the documentation in `repositoryTypes` down to the [Supported repository transport protocols](https://docs.gradle.org/current/userguide/repository_types.html#sub:supported_transport_protocols) section, add "HTTP header" to the table rows for HTTP and HTTPS there, link to the DSL reference for both types etc.

Please let me know if that's clear enough. If you don't feel up for it, I could take it from here. Just say the word. 🙂

Reply:
@marcphilipp, thanks. I will work on it this weekend when I find the time, then I will be out and offline for a week. But I'd like to dive into it to learn more about the gradle code.

Reply:
@marcphilipp, I have problems removing the interface: I have to give the fully qualified name in the `build.gradle`:

```
credentials(org.gradle.api.credentials.HttpHeaderCredentials) {
            name = "TestHttpHeaderName"
            value = "TestHttpHeaderValue"
        }
        authentication { header(HttpHeaderAuthentication) }
```

See https://github.com/paulwellnerbou/gradle/blob/5571-http-header-auth-remove-interface-move-Default-classes/subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/http/HttpAuthenticationDependencyResolutionIntegrationTest.groovy#L177.

But this seems to be a test issue only. It works in a real world gradle project.

Reply:
Besides that, I think I completed all missing points you mentioned. I documented the authentication at the bottom of the page, under *Configuring HTTP authentication schemes*:

https://github.com/paulwellnerbou/gradle/blob/5571-http-header-authentification-support/subprojects/docs/src/docs/userguide/repositoryTypes.adoc